crMeta/crMetadata.go

```diff
@@ -282,9 +340,10 @@ func ResolveConflictByCAS(req *base.WrappedMCRequest, resp *mc.MCResponse, specs
 	if doc_meta_target.Cas > doc_meta_source.Cas {
 		sourceWin = false
 	} else if doc_meta_target.Cas == doc_meta_source.Cas {
-		if doc_meta_target.RevSeq > doc_meta_source.RevSeq {
+		// We cannot use revId if they are not saved (set to 0) which is the case for import mutation.
+		if doc_meta_target.RevSeq > doc_meta_source.RevSeq && doc_meta_source.RevSeq != 0 && doc_meta_target.RevSeq != 0 {
 			sourceWin = false
-		} else if doc_meta_target.RevSeq == doc_meta_source.RevSeq {
+		} else if doc_meta_target.RevSeq == doc_meta_source.RevSeq || doc_meta_source.RevSeq == 0 || doc_meta_target.RevSeq == 0 {
```

Comment:
> This is what I'm reading for RevID based CR.
Do you mean Cas based CR? because this is in ResolveConflictByCAS().

If either mutation is import mutation, then we skip comparing revId. The rest of the algorithm is not changed. Delete will win if Cas is the same.

As for flags, in one of the meetings, I proposed to skip flags and only compare CAS for import mutations because of Adam's comment on flags. John responded saying something like "how about we keep it the same since there is no guarantee that the flags will change". This was said in our scrum meeting, Adam was not there.

Reply:
OK makes sense. Thanks for the refresh.